lib/smart_listing.rb

```diff
@@ -109,7 +109,7 @@ def setup params, cookies
       else
         # let's sort by all attributes
         #
-        @collection = @collection.order(sort_keys.collect{|s| "#{s[1]} #{@sort[s[0]]}" if @sort[s[0]]}.compact) if @sort && !@sort.empty?
+        @collection = @collection.order(sort_keys.collect{|s| Arel.sql("#{s[1]} #{@sort[s[0]]}") if @sort[s[0]]}.compact) if @sort && !@sort.empty?
```

Comment:
This doesn't seem related to Ruby 3, do you know what it is for?

Reply:
[Here's an article](https://dev.to/ashawareb/arel-and-ruby-on-rails-58m3) with some more information. It looks like it was introduced in Ruby 3.